packages/expo-modules-core/ios/JSI/JavaScriptValue.swift

```diff
@@ -72,7 +72,7 @@ public extension JavaScriptValue {
     throw JavaScriptValueConversionException((kind: kind, target: "Array"))
   }
 
-  func asDict() throws -> [String: Any?] {
+  func asDict() throws -> [String: Any] {
```

Comment:
`Any` also accepts `nil` values, so the additional `?` was only causing problems in unwrapping the optional